docs/user_guide/customizing_dependencies/imagespec.md

```diff
@@ -103,3 +103,9 @@ or
 image = ImageSpec(registry="ghcr.io/flyteorg", packages=["pandas"]).force_push()
 ```
 [flytesnacks]: https://github.com/flyteorg/flytesnacks/tree/master/examples/customizing_dependencies/
+
+## Getting source files into ImageSpec
+Typically, getting source code files into a task's image at run time on a live Flyte backend is done through the fast registration mechanism.
```

Comment:
```suggestion
Typically, getting source code files into a task's image at run time on a live Flyte backend is done through the fast registration mechanism.

However, if your `ImageSpec` constructor specifies a `source_root` and the `copy` argument is set to something other than `CopyFileDetection.NO_COPY`, then files will be copied regardless of fast registration status.
```

Reply:
I think it's fine to just say the same thing in two places for clarity. In the future, we can use reusable text snippets for stuff like this.